File: src/front/myb.front/libs/doc-management-module/src/lib/models/Folder.ts
import { BaseModel } from '../../../../shared/infra/models/base.model';
import { DocumentModel } from "./DocumentModel";

export interface Folder extends BaseModel {
  
    parentId: number | null; 
    folderName: string;
    createdBy: string;
    editedBy: string;
    documents?: DocumentModel[] | null;
   // children?: Folder[] | null; 
    
  
  }

File: src/services/document-management/Myb.document.Model/Folder.cs
using Myb.Common.Models;
using Myb.UserManager.Models;


namespace Myb.document.Model
{
    public class Folder : BaseEntity
    {
    // public int? Id { get; set; }
        public int? ParentId { get; set; }
        public string FolderName { get; set; }

        public  virtual ICollection<DocumentModel> ?Documents { get; set; } 

        public string CreatedBy { get; set; }
        public string EditedBy { get; set; }

        public string? UserId { get; set; }
        public int? CompanyId { get; set; }



    }
}
